plugins/chat/app/services/chat/update_message.rb

```diff
@@ -30,34 +31,38 @@ class UpdateMessage
 
     class Contract
       attribute :message_id, :string
+      validates :message_id, presence: true
+
       attribute :message, :string
+      validates :message, presence: true, if: -> { upload_ids.blank? }
+
       attribute :upload_ids, :array
 
-      validates :message_id, presence: true
-      validates :message, presence: true, if: -> { upload_ids.blank? }
+      attribute :process_inline, :boolean, default: Rails.env.test?
     end
 
     private
 
+    def enforce_system_membership(guardian:, message:, **)
+      message.chat_channel.add(guardian.user) if guardian.user.is_system_user?
+    end
+
     def fetch_message(contract:, **)
-      ::Chat::Message
-        .strict_loading
-        .includes(
-          :chat_mentions,
-          :bookmarks,
-          :chat_webhook_event,
-          :uploads,
-          :revisions,
-          reactions: [:user],
-          thread: [:channel, last_message: [:user]],
-          chat_channel: [
-            :last_message,
-            :chat_channel_archive,
-            chatable: [:topic_only_relative_url, direct_message_users: [user: :user_option]],
-          ],
-          user: :user_status,
-        )
```

Comment:
I decided to remove this for now as it was doing more harms than good, I will think of a better solution

Reply:
It was raising in some specs. And I feel like it doesn't matter that much in update which is called somewhat rarely and not really a hot path.

I still want to improve it, but I want to investigate better solution, maybe a mix of includes and autoload. Not sure yet.